pyproject.toml

```diff
@@ -0,0 +1,33 @@
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[project]
+authors = [
+    {name = "Tom Hu", email = "thomas.hu@sentry.io"},
+    {name = "Codecov Support", email = "support@codecov.io"},
+]
+description = "Codecov Command Line Interface"
+requires-python = ">= 3.9"
+dependencies = [
+    "click==8.*",
+    "httpx==0.27.*",
+    "ijson==3.*",
+    "pyyaml==6.*",
+    "regex",
+    "responses==0.21.*",
+    "sentry-sdk>=2.20.0",
+    "test-results-parser==0.5.*",
+    "tree-sitter==0.20.*",
+    "wrapt>=1.17.2",
+]
+license = {file = "LICENSE"}
+name = "codecov-cli"
+readme = "README.md"
+version = "10.0.1"
```

Comment:
I prefer generating the version via `setuptools-scm` but that's out of the scope of the initial refactor. Can be discussed separately.